src/main/resources/GoogleApps/JSExtension.xml

```diff
@@ -111,50 +122,212 @@
         <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
       </use>
     </class>
+    <property>
+      <cache>short</cache>
+    </property>
+    <property>
+      <code>define('googleapps-buttons', {
+  prefix: 'googleapps.',
+  keys: [
+    'edit.editingoogleapps.link',
+    'import.importfromgoogleapps'
+  ]
+});
+require(['jquery', 'xwiki-l10n!googleapps-buttons'], function ($, l10n) {
+  var listener = function () {
+    var hasPagination = $("#_attachments table").length &gt; 0;
+    $("#_attachments span.name").each(function (i, elem) {
+      var searchPoint = hasPagination ? $(elem).closest('tr').find('.attachmentActions') : $(elem.parentNode).find('.xwikibuttonlinks');
+      if ($(searchPoint).find('a.editlink').length) {
+        return;
+      }
+      var dlink = hasPagination ? $(searchPoint).find(".actiondelete") : $(searchPoint).find(".deletelink");
+      if (dlink.length &gt; 0) {
+        var dlinkurl = dlink[0].href;
+        var dlinkpos = dlinkurl.indexOf("?");
+        var filename = dlinkurl.substring(dlinkurl.lastIndexOf("/", dlinkpos) + 1, dlinkpos);
+        var googleDoc = new XWiki.Document('EditInGoogleApps', 'GoogleApps');
+        var queryParams = {
+          page: encodeURIComponent(XWiki.currentSpace) + '.' + encodeURIComponent(XWiki.currentPage),
+          name: filename
+        };
+        searchPoint.append('&lt;a class="editlink" href="' + googleDoc.getURL('view', $.param(queryParams)) + '" title="' + l10n.get('edit.editingoogleapps.link') + '"&gt;' + l10n.get('edit.editingoogleapps.link') + '&lt;/a&gt;');
```

Comment:
It's never a good idea to inject values in HTML with string concatenation.

```suggestion
        $('&lt;a&gt;&lt;/a&gt;').attr({
          class: 'editlink',
          href: googleDoc.getURL('view', $.param(queryParams)),
          title: l10n.get('edit.editingoogleapps.link')
        }).text(l10n.get('edit.editingoogleapps.link')).appendTo(searchPoint);
```

Reply:
It is also very neat, clean looking.